web/client/reducers/geoProcessing.js

```diff
@@ -181,7 +183,8 @@ function geoProcessing( state = {
             ...state,
             [action.source]: {
                 ...state[action.source],
-                features: (state[action.source].features || []).concat(action.data.features || []),
+                features: (state[action.source].features || []).concat(action.data.features || []).filter(f => !f?.properties?.measureId).map((ft, i)  => ({...ft,
+                    id: createFeatureId(ft, i)})),
```

Comment:
Please explain why you are filtering out all features that do not have `measureId` and creating ids for all features. It is not trivial the reason and it is  a special behaviour that have to be commented.
I suggest to do this changes outside the reducer. The reducer should expect well formed features.

Reply:
the main problem was that the vector layer generated from feature measurements are missing ids and it makes some other part of the code to fail. i'll centralize this behaviour in the layers selector of the geoprocessing tool and also in epics